I am surprised by the PR testing issue:

```
[2026/03/18 17:21:05.441] In file included from wiredtiger/test/format/alter.c:29:
[2026/03/18 17:21:05.441] wiredtiger/test/format/format.h:106:10: error: multiple candidates for header 'config.h' found; directory 'wiredtiger/test/format' chosen, ignoring others including 'wiredtiger/src/include' [-Werror,-Wshadow-header]
[2026/03/18 17:21:05.447]   106 | #include "config.h"
[2026/03/18 17:21:05.447]       |  
```

Looks like an infra issue, it's impacting my other patch builds.

--- a/src/conn/conn_layered_page_log.c
+++ b/src/conn/conn_layered_page_log.c
@@ -516,6 +516,9 @@ __disagg_put_meta(WT_SESSION_IMPL *session, uint64_t page_id, const WT_ITEM *ite
     return (0);
 }
 
+/* Limit the amount of bytes we dump in the metadata page dump. */
+#define WT_DISAGG_META_DUMP_MAX 1024
+
 /*
  * __wti_disagg_fetch_shared_meta --
  *     Fetch the checkpoint metadata page, validate it, and return a zero-terminated buffer copy.
@@ -536,12 +539,14 @@ __wti_disagg_fetch_shared_meta(
     if (ckpt_meta->has_metadata_checksum) {
         const uint32_t checksum = __wt_checksum(item->data, item->size);
         if (checksum != ckpt_meta->metadata_checksum) {
+            const size_t dump_size =
+              item->size > WT_DISAGG_META_DUMP_MAX ? WT_DISAGG_META_DUMP_MAX : item->size;
             WT_ERR(__wt_scr_alloc(session, 0, &hex_buf));
             WT_ERR_MSG(session, EIO,
               "Checkpoint metadata corruption detected: lsn=%" PRIu64 ", expected=0x%" PRIx32
               ", got=0x%" PRIx32 ", data=[%s]",
               ckpt_meta->metadata_lsn, ckpt_meta->metadata_checksum, checksum,
-              __wt_buf_set_printable(session, item->data, item->size, false, hex_buf));
+              __wt_buf_set_printable(session, item->data, dump_size, false, hex_buf));
         }
     }
 

--- a/test/suite/test_layered64.py
+++ b/test/suite/test_layered64.py
@@ -105,4 +105,4 @@ class test_layered64(wttest.WiredTigerTestCase):
         self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
             lambda: self.conn.reconfigure(
                 f'disaggregated=(checkpoint_meta="{corrupted_checkpoint_meta}")'),
-            "/Checkpoint metadata checksum mismatch/")
+            "/Checkpoint metadata corruption detected/")

